actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -95,7 +95,11 @@ def variant=(variant)
       end
 
       def variant
-        @variant ||= ActiveSupport::ArrayInquirer.new
+        @variant ||= if parameters.key? :variant
+                       public_send(:variant=, parameters[:variant].try(:to_sym))
```

Comment:
Can't we use `self.variant =` here? We also try to avoid `try` internally in the framework, let's remove it by checking for a value in `parameters[:variant]` instead of there being a key. Like this:

```ruby
def variant
  @variant ||=
    if parameters[:variant]
      self.variant = parameters[:variant].to_sym
    else
      ActiveSupport::ArrayInquirer.new
    end
end